core/storage/src/relational/fetch.rs

```diff
@@ -288,148 +292,183 @@ impl RelationalStorage {
+            .query_simple_block_by_number(
+                block_number.try_into().map_err(|_| DBError::WrongHeight)?,
+            )
+            .await?;
+        self.get_simple_block(block_hash, block_number, parent_hash, block_timestamp)
+            .await
     }
 
     pub(crate) async fn get_simple_block_by_block_hash(
         &self,
         block_hash: H256,
     ) -> Result<SimpleBlock> {
-        let block_table = self
-            .query_block_by_hash(to_rb_bytes(block_hash.as_bytes()))
+        let (block_hash, block_number, parent_hash, block_timestamp) = self
+            .query_simple_block_by_hash(block_hash.as_bytes())
             .await?;
-        self.get_simple_block(&block_table).await
+        self.get_simple_block(block_hash, block_number, parent_hash, block_timestamp)
+            .await
     }
 
-    async fn get_simple_block(&self, block_table: &BlockTable) -> Result<SimpleBlock> {
-        let txs = self
-            .query_transactions_by_block_hash(&block_table.block_hash)
+    async fn get_simple_block(
+        &self,
+        block_hash: H256,
+        block_number: BlockNumber,
+        parent_hash: H256,
+        timestamp: u64,
+    ) -> Result<SimpleBlock> {
+        let transactions = self
+            .query_transaction_hashes_by_block_hash(block_hash.as_bytes())
             .await?;
         Ok(SimpleBlock {
-            block_number: block_table.block_number,
-            block_hash: rb_bytes_to_h256(&block_table.block_hash),
-            parent_hash: rb_bytes_to_h256(&block_table.parent_hash),
-            timestamp: block_table.block_timestamp,
-            transactions: txs
-                .iter()
-                .map(|tx| rb_bytes_to_h256(&tx.tx_hash))
-                .collect::<Vec<H256>>(),
+            block_number,
+            block_hash,
+            parent_hash,
+            timestamp,
+            transactions,
         })
     }
 
     pub(crate) async fn query_scripts(
         &self,
-        script_hashes: Vec<RbBytes>,
-        code_hash: Vec<RbBytes>,
+        script_hashes: Vec<H160>,
+        code_hashes: Vec<H256>,
         args_len: Option<usize>,
-        args: Vec<RbBytes>,
+        args: Vec<Bytes>,
     ) -> Result<Vec<packed::Script>> {
-        if script_hashes.is_empty() && code_hash.is_empty() && args_len.is_none() && args.is_empty()
-        {
+        if script_hashes.is_empty() && args.is_empty() {
             return Err(DBError::InvalidParameter(
                 "no valid parameter to query scripts".to_owned(),
             )
             .into());
         }
 
-        let mut wrapper = self.pool.wrapper();
-
+        // build query str
+        let mut query_builder = SqlBuilder::select_from("mercury_script");
+        let mut query = query_builder.field("script_code_hash, script_args, script_type");
         if !script_hashes.is_empty() {
-            wrapper = wrapper.in_array("script_hash_160", &script_hashes)
+            query = query.and_where_in(
```

Comment:
SQLx currently only provides the function of directly binding arrays to postgres. 
Considering that the unit test uses sqlite as the embedded database, the general method is temporarily selected. 
SQLx mentioned that it will support all databases for array binding in the future, maybe wait and see?